app/components/forms/events/view/create-access-code.js

```diff
@@ -0,0 +1,89 @@
+import Ember from 'ember';
+import FormMixin from 'open-event-frontend/mixins/form';
+
+const { Component } = Ember;
+
+export default Component.extend(FormMixin, {
+  getValidationRules() {
+    return {
+      inline : true,
+      delay  : false,
+      on     : 'blur',
+
+      fields: {
+        accessCode: {
+          identifier : 'access_code',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.l10n.t('Please enter access code')
+            }
+          ]
+        },
+        numberOfAccessTickets: {
+          identifier : 'number_of_access_tickets',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.l10n.t('Please enter number of tickets')
+            },
+            {
+              type   : 'number',
+              prompt : this.l10n.t('Please enter proper number of tickets')
+            }
+          ]
+        },
+        allTicketTypes: {
+          identifier : 'all_ticket_types',
+          rules      : [
+            {
+              type   : 'checked',
+              propmt : this.l10n.t('Please select the appropriate choices')
+            }
+          ]
+        },
+        min: {
+          identifier : 'min',
+          optional   : true,
+          rules      : [
+            {
+              type   : 'number',
+              propmt : this.l10n.t('Please enter the proper number')
+            }
+          ]
+        },
+        max: {
+          identifier : 'max',
+          optional   : true,
+          rules      : [
+            {
+              type   : 'number',
+              propmt : this.l10n.t('Please enter the proper number')
+            }
+          ]
+        },
+        accessLink: {
+          identifier : 'access_link',
+          optional   : true,
+          rules      : [
+            {
+              type   : 'url',
+              propmt : this.l10n.t('Please enter a valid url')
+            }
+          ]
+        }
+      }
+    };
+  },
+
+  actions: {
+    generateAccessLink(value) {
+      let accesslink = `https://open-event-dev.herokuapp.com/e/event_id/?code=${value}`;
```

Comment:
@fossasia/open-event-frontend 👆 this will work inside all components. We have injected the routing service globally into all components (_It's a private internal service of ember and we should really be using it. But until ember exposes a proper routing service to use inside components, we don't really have much choice on that_) . But don't try the same on a controller or a route. They already have a different router service.